ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/observer/UtxoLedgerTokenStateObserver.kt

```diff
@@ -0,0 +1,76 @@
+package net.corda.v5.ledger.utxo.observer
+
+import net.corda.v5.ledger.utxo.ContractState
+import net.corda.v5.ledger.utxo.StateAndRef
+import net.corda.v5.ledger.utxo.token.selection.TokenSelection
+
+/**
+ * The [UtxoLedgerTokenStateObserver] observes produced contract states of type [T] when they are committed to the
+ * ledger. Users should implement this interface for any states that need to be selectable via the [TokenSelection] API
+ *
+ * Example of use in Java.
+ * ```Java
+ * public class ExampleStateJ implements ContractState {
+ *     public List<PublicKey> participants;
+ *     public SecureHash issuer;
+ *     public String currency;
+ *     public BigDecimal amount;
+ *
+ *     @NotNull
+ *     @Override
+ *     public List<PublicKey> getParticipants() {
+ *         return participants;
+ *     }
+ * }
+ *
+ * class UtxoLedgerTokenStateObserverKotlinExample : UtxoLedgerTokenStateObserver<ExampleState> {
+ *
+ *     override val stateType =  ExampleState::class.java
+ *
+ *     override fun onProduced(stateAndRef: StateAndRef<ExampleState>): UtxoToken {
+ *         val state = stateAndRef.state.contractState
+ *         return UtxoToken(
+ *             UtxoTokenPoolKey(ExampleState::class.java.typeName, state.issuer, state.currency),
+ *             state.amount,
+ *             UtxoTokenFilterFields()
+ *         )
+ *     }
```

Comment:
This claims to be a Java example but looks a bit kotlinesque to me.

Reply:
fixed.